src/tcp_tunnel_tools.js

```diff
@@ -1,11 +1,11 @@
 const net = require('net');
 const TcpTunnel = require('./tcp_tunnel');
-const { parseUrl, nodeify } = require('./tools');
+const { nodeify } = require('./utils/nodeify');
 
 const runningServers = {};
 
 function createTunnel(proxyUrl, targetHost, providedOptions = {}, callback) {
-    const parsedProxyUrl = parseUrl(proxyUrl);
+    const parsedProxyUrl = new URL(proxyUrl);
     if (!parsedProxyUrl.hostname || !parsedProxyUrl.port) {
```

Comment:
looking at the `parseUrl`, it automatically added default ports, while `new URL` will keep an empty string if port is not explicitly provided, so I'd say this validation can be triggered for portless urls now?

Reply:
`TcpTunnel` will be another PR

Reply:
but you broke it in this one, right? so i guess at least add a todo

Reply:
The todo is in the PR description, but fine, I'll remove it (it's moot since the protocol check is enough).

This is the wrong PR to fix things like this.